app.ts

```diff
@@ -0,0 +1,406 @@
+						this.currentPage = Math.floor((endID + 1) / (maxRecords + 1));
+						pageEnd = Math.ceil(this.currentPage / 10) * 10;
+						pageStart = pageEnd - 9;
+						if (endID >= count) {
+							startID = ((this.currentPage - 1) * maxRecords) + 1;
+							endID = (count - 1);
+							this.paginationStart = pageStart;
+							this.paginationEnd = pageEnd;
+						}
+					} else {
+						throw new Error("Please provide a valid integer.");
+					}
+				})
+				.then(() => {
+					return this.pageNumbers(pageStart, pageEnd);
+				})
+				.then(() => {
+					return this.showRecords(startID, endID);
+				})
+				.then(() => {
+					$('.results-select').prop('disabled', false);
+				})
+				.catch(error => {
+					throw new Error(`Failed when clicking on the search range: ${error}`);
+				});
+		});
+	}
+
+	/** Will calculate the amount records to be shown according to the screen height. */
+	adjustDisplayedRecords(): Promise<void> {
+		let screenHeight = <number>($('#records-table-body').height());
+		let pageStart: number;
+		let pageEnd: number;
+		return this.getRecordCount()
+			.then(count => {
+				let maxRecords = this.recordsPerPage();
+				let inputNumber = this.input();
+				if (inputNumber === -1) {
+					let newCurrentPage = Math.ceil(this.currentFromID / maxRecords);
+					if (newCurrentPage === 0) {
+						this.currentFromID = 0;
+						newCurrentPage = 1;
+					}
+					this.currentToID = this.currentFromID + maxRecords;
+					this.currentPage = newCurrentPage;
+					let originalID = ((this.currentPage * maxRecords) + (this.currentPage - 1)) - maxRecords;
+					this.difference = originalID - this.currentFromID;
+				} else {
+					if (this.currentToID >= count) {
+						this.currentToID = (count - 1);
+					}
+					let newCurrentPage = Math.ceil(inputNumber / maxRecords);
+					this.currentToID = newCurrentPage * maxRecords;
+					this.currentPage = newCurrentPage;
+					this.currentFromID = (this.currentPage - 1) * maxRecords + 1;
+				}
+				if (this.currentFromID === 0) {
+					let divisor = screenHeight < 136 ? maxRecords : maxRecords - 1;
+					pageEnd = Math.ceil(Math.floor(this.currentToID / divisor) / 10) * 10;
+				} else {
+					pageEnd = Math.ceil(Math.floor(this.currentToID / maxRecords) / 10) * 10;
+				}
+				pageStart = pageEnd - 9;
+				this.paginationStart = pageStart;
+				this.paginationEnd = pageEnd;
+				$("#records-table-body").empty();
+				return this.showRecords(this.currentFromID, this.currentToID);
+			})
+			.then(() => {
+				return this.pageNumbers(pageStart, pageEnd);
+			})
+			.catch(error => {
+				throw new Error(`Failed when resizing the window: ${error}`);
+			});
+	}
+
+	/** When resizing the window. Timeout is put in place so that the function doesn't
+	 * take in every value returned during resizing. */
+	resize(): void {
+		if (this.resizeTimer !== null) {
+			clearTimeout(this.resizeTimer);
+		}
+		this.resizeTimer = setTimeout(() => {
+			return this.adjustDisplayedRecords()
```

Comment:
Returning here does not make a lot of sens. `setTimeout` can't do anything with your promise.